watchtower/lookout/lookout.go

```diff
@@ -118,9 +165,18 @@ func (l *Lookout) watchBlocks(epochs *chainntnfs.BlockEpochEvent) {
 			log.Debugf("Fetching block for (height=%d, hash=%s)",
 				epoch.Height, epoch.Hash)
 
-			// Fetch the full block from the backend corresponding
-			// to the newly arriving epoch.
-			block, err := l.cfg.BlockFetcher.GetBlock(epoch.Hash)
+			// Fetch the full block corresponding to the newly
+			// arriving epoch from the backend.
+			// This is done using the fetchBlockWithRetries
+			// function, which attempts to fetch the block
+			// multiple times in case of temporary failures.
+			// This operation is crucial for keeping the local
+			// representation of the blockchain up-to-date with
+			// the newly arriving epoch, ensuring the system remains
+			// in sync with the backend.
+			block, err := fetchBlockWithRetries(l.cfg.BlockFetcher,
+				epoch.Hash, 3)
```

Comment:
nit: formatting 

block, err := fetchBlockWithRetries(
    l.cfg.BlockFetcher, epoch.Hash, 3, 
)

Reply:
also - you could probably make `fetchBlockWithRetries` a method instead since then there is no need to pass the block fetcher. 

Then you can also more easily listen on the quit channel between retries

Reply:
> also - you could probably make `fetchBlockWithRetries` a method instead since then there is no need to pass the block fetcher.
> 
> Then you can also more easily listen on the quit channel between retries

Could you help me understand a bit more about your approach? How exactly are we planning to keep an eye on the quit channel? It kind of sounds like you're suggesting that every channel needs constant monitoring for breaches. Maybe I'm missing something here. Can you break it down a bit more for me? Thanks!

Reply:
ah - so the channel im referring to here is a golang channel, not a lightning channel.  So I mean the `Lookout`'s `quit chan struct` member. It will signal on that channel when it is exiting.